### PR TITLE
Set source handler to workers

### PR DESCRIPTION
**Pull request by pubudu91**
This PR sets the source handler to the contexts of the workers.

### CODE REVIEW

**Comment by pubudu91** [comment]
@chanakaudaya Requested changes incorporated.

File: modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java [hunk 1]
```diff
@@ -150,8 +151,13 @@ public static void  invokeResource(CarbonMessage cMsg, CarbonCallback callback,
...
         context.setServiceInfo(serviceInfo);
         context.setCarbonMessage(resourceMessage);
-        context.setProperty("SRC_HANDLER", resourceMessage.getProperty("SRC_HANDLER"));
         context.setBalCallback(new DefaultBalCallback(resourceCallback));
+
+        SourceHandler srcHandler = null;
+        if (resourceMessage.getProperty(Constants.SRC_HANDLER) != null) {
+            srcHandler = (SourceHandler) resourceMessage.getProperty(Constants.SRC_HANDLER);
+            context.setProperty(Constants.SRC_HANDLER, srcHandler);
+        }
```
Comment (shafreenAnfar) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java]: Now that we set the SH here, I suppose we don't have to set it below

https://github.com/ballerinalang/ballerina/blob/master/modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/actions/http/AbstractHTTPAction.java#L144

Comment (pubudu91) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java]: Didn't get the reason behind the conditional check in the link you posted. If the source handler is not there in the message, it's definitely there in the context?

Comment (muthulee) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java]: I think what Shafreen means is line number 144 is not needed, but line number 146 is needed.

**Pull request merged**